configure-android.sh

```diff
@@ -1,199 +1,223 @@
+
+ANDROID_TOOLCHAIN=$ANDROID_NDK_ROOT/toolchains/llvm/prebuilt/$BUILD_MACHINE_OS_TYPE-$BUILD_MACHINE_OS_ARCH
+
+#########################################################################################
+
+cat <<EOF
+BUILD_MACHINE_OS_TYPE=$BUILD_MACHINE_OS_TYPE
+BUILD_MACHINE_OS_ARCH=$BUILD_MACHINE_OS_ARCH
+
+ANDROID_NDK_ROOT =$ANDROID_NDK_ROOT
+ANDROID_API_LEVEL=$ANDROID_API_LEVEL
+ANDROID_TOOLCHAIN=$ANDROID_TOOLCHAIN
+
+TARGET_TRIPLE=$TARGET_TRIPLE
+
+LIBPCAP_INCLUDE_DIR=$LIBPCAP_INCLUDE_DIR
+LIBPCAP_LIBRARY_DIR=$LIBPCAP_LIBRARY_DIR
+EOF
+
+#########################################################################################
+
+unset TEMP_DIR
+unset TEMP_CONFIG_FILE
+
+TEMP_DIR=$(mktemp -d)
+TEMP_CONFIG_FILE="$TEMP_DIR/config.mk"
+
+cat > "$TEMP_CONFIG_FILE" <<EOF
+PCAPPLUSPLUS_HOME   = $PWD
+ANDROID_TARGET      = $TARGET_TRIPLE$ANDROID_API_LEVEL
+ANDROID_NDK_PATH    = $ANDROID_NDK_ROOT
+ANDROID_API_VERSION = $ANDROID_API_LEVEL
+EOF
 
-# copy the basic Android platform.mk
-cp -f mk/platform.mk.android $PLATFORM_MK
+#########################################################################################
 
-# copy the common (all platforms) PcapPlusPlus.mk
-cp -f mk/PcapPlusPlus.mk.common $PCAPPLUSPLUS_MK
+PCAPPLUSPLUS_MK="$SCRIPT_DIR/mk/PcapPlusPlus.mk"
 
-# add the Android definitions to PcapPlusPlus.mk
-cat mk/PcapPlusPlus.mk.android >> $PCAPPLUSPLUS_MK
+cp -f "$TEMP_CONFIG_FILE"                       "$PCAPPLUSPLUS_MK"
+cat "$SCRIPT_DIR/mk/PcapPlusPlus.mk.common"  >> "$PCAPPLUSPLUS_MK"
+cat "$SCRIPT_DIR/mk/PcapPlusPlus.mk.android" >> "$PCAPPLUSPLUS_MK"
 
-# set current directory as PCAPPLUSPLUS_HOME in platform.mk
-echo -e "\nPCAPPLUSPLUS_HOME := "$PWD >> $PLATFORM_MK
+cat >> "$PCAPPLUSPLUS_MK" <<EOF
+LIBPCAP_INCLUDE_DIR := $LIBPCAP_INCLUDE_DIR
+PCAPPP_INCLUDES += -I\$(LIBPCAP_INCLUDE_DIR)
+LIBPCAP_LIB_DIR := $LIBPCAP_LIBRARY_DIR
+PCAPPP_LIBS_DIR += -L\$(LIBPCAP_LIB_DIR)
+EOF
 
-# set target variable in PcapPlusPlus.mk
-sed -i "1s|^|ANDROID_TARGET := $TARGET$API\n\n|" $PCAPPLUSPLUS_MK
+#########################################################################################
 
-# set NDK path variable in PcapPlusPlus.mk
-sed -i "1s|^|ANDROID_NDK_PATH := $NDK_PATH\n\n|" $PCAPPLUSPLUS_MK
+PLATFORM_MK="$SCRIPT_DIR/mk/platform.mk"
 
-# set current direcrtory as PCAPPLUSPLUS_HOME in PcapPlusPlus.mk
-sed -i "1s|^|PCAPPLUSPLUS_HOME := $PWD\n\n|" $PCAPPLUSPLUS_MK
+cp -f "$TEMP_CONFIG_FILE"                    "$PLATFORM_MK"
+cat "$SCRIPT_DIR/mk/platform.mk.android"  >> "$PLATFORM_MK"
 
-# set target variable in platform.mk
-sed -i "1s|^|ANDROID_TARGET := $TARGET$API\n\n|" $PLATFORM_MK
+#########################################################################################
 
-# set NDK path variable in platform.mk
-sed -i "1s|^|ANDROID_NDK_PATH := $NDK_PATH\n\n|" $PLATFORM_MK
+sed -i "/ANDROID_TOOLCHAIN=/c ANDROID_TOOLCHAIN=$ANDROID_TOOLCHAIN" "$PLATFORM_MK"
+sed -i "/ANDROID_TOOLCHAIN=/c ANDROID_TOOLCHAIN=$ANDROID_TOOLCHAIN" "$PCAPPLUSPLUS_MK"
```

Comment:
Currently the `ANDROID_TOOLCHAIN`  variable is defined in [`PcapPlusPlus.mk.android`](https://github.com/seladb/PcapPlusPlus/blob/5fd8db1dae9fd8395395e82a1d19017d60a0e632/mk/PcapPlusPlus.mk.android#L1). Maybe instead of replacing its value we can just remove it from there and add it here to `TEMP_CONFIG_FILE`?